### ISSUE TITLE
Add information about requires/build_requires to lockfiles

### ISSUE THREAD

**Issue by jgsogo** [conan-io/conan#5673]
Currently, all `requires` and `build_requirers` are listed in the lockfile under the same JSON node

```json
...
   "80ee61f4-c8ae-11e9-8476-0242ac130007": {
    "pref": "core-messages/0.0@sword/sorcery#58f0fd53713d6163f6c9a1eb620e626a:bbeef259ed68d0bfcdfe6c6e7c5f8d3581876931#ce16a30bb3eff98e4dd0fea966ef3942",
    "options": "shared=False\nprotobuf:fPIC=True\nprotobuf:lite=False\nprotobuf:shared=False\nprotobuf:with_zlib=False",
    "modified": true,
    "requires": {
     "protobuf/3.6.1@bincrafters/stable#0": "80ee5af6-c8ae-11e9-8476-0242ac130007",
     "protoc_installer/3.6.1@bincrafters/stable": "80ee5ef2-c8ae-11e9-8476-0242ac130007"
    }
...
```

it would be useful to keep that information so we can use it to populate the BuildInfo for Artifactory. We can associate each group to a different `scope` and it could be used during a promotion to discriminate actual dependencies from tools used to build.

**Comment by uilianries** [comment]
Graph lock uses a list of nodes the create the list of requirements. When filtering build requirements, it adds to the general list with no distinction:

https://github.com/conan-io/conan/blob/3953894b5ee63e76086f7880efaa5032e5974127/conans/client/graph/graph_manager.py#L252

Or we need to add an extra information in Node class, to make that distinction, or we could create a separated list of Node for build requirements.

**Comment by memsharded** [comment]
This was released in Conan 1.22, implemented in https://github.com/conan-io/conan/pull/6237
